src/app/clusters/ota-requestor/GenericOTARequestorDriver.cpp

```diff
@@ -48,6 +48,8 @@ using namespace app::Clusters::OtaSoftwareUpdateRequestor::Structs;
 constexpr uint32_t kDelayQueryUponCommissioningSec = 30; // Delay before sending the initial image query after commissioning
 constexpr uint32_t kImmediateStartDelaySec         = 1;  // Delay before sending a query in response to UrgentUpdateAvailable
 constexpr System::Clock::Seconds32 kDefaultDelayedActionTime = System::Clock::Seconds32(120);
+System::Clock::Seconds32 mDelayNotifyUpdateAppliedSeconds =
```

Comment:
Why is there a delay to notify? This isn't part of the spec. Is there a particular reason ESP32 needs to have a delay prior to notifying?

Reply:
Yes, after reboot wifi takes some time to start.

Reply:
This routine is executed very early and Wi-Fi is not yet connected at that point, so buying some time till the wifi connects.

Reply:
Wondering if this delay should be done at the application level then?

Reply:
I'm wondering if `GenericOTARequestorDriver::Init` should just be delayed from the application level instead?

Reply:
Yes, I think that's probably the right approach - create some kind of callback for post-connection to network. The notify will be automatically done in `DefaultOTARequestorDriver::Init` if on first boot from an upgrade anyway so perhaps the callback (indicating W-Fi is connected) is the one responsible for calling all the OTA related initialization code? Even in non-upgrade scenario, we probably should just wait for Wi-Fi to be connected before all the inits are called, right?

Reply:
@carol-apple @dhrishi accordingly made changes please review.